pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -401,21 +397,38 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 		message := fmt.Sprintf("PipelineRun %s/%s awaiting remote resource", pr.Namespace, pr.Name)
 		pr.Status.MarkRunning(ReasonResolvingPipelineRef, message)
 		return nil
-	case errors.Is(err, trustedresources.ErrResourceVerificationFailed):
-		message := fmt.Sprintf("PipelineRun %s/%s referred pipeline failed signature verification", pr.Namespace, pr.Name)
-		pr.Status.MarkFailed(ReasonResourceVerificationFailed, message)
-		return controller.NewPermanentError(err)
 	case err != nil:
 		logger.Errorf("Failed to determine Pipeline spec to use for pipelinerun %s: %v", pr.Name, err)
 		pr.Status.MarkFailed(ReasonCouldntGetPipeline,
 			"Error retrieving pipeline for pipelinerun %s/%s: %s",
 			pr.Namespace, pr.Name, err)
 		return controller.NewPermanentError(err)
-	default:
-		// Store the fetched PipelineSpec on the PipelineRun for auditing
-		if err := storePipelineSpecAndMergeMeta(ctx, pr, pipelineSpec, pipelineMeta); err != nil {
-			logger.Errorf("Failed to store PipelineSpec on PipelineRun.Status for pipelinerun %s: %v", pr.Name, err)
+	}
+
+	// Skip the verification if it is not inline pipeline;
+	if pr.Spec.PipelineSpec == nil && pr.Status.PipelineSpec == nil {
+		var pipelineToBeVerified *v1beta1.Pipeline
+		if pipelineMeta != nil {
+			pipelineToBeVerified = &v1beta1.Pipeline{
+				ObjectMeta: *pipelineMeta.ObjectMeta,
+				Spec:       *pipelineSpec,
```

Comment:
These could panic if they're nil (same with taskToBeVerified below)

Reply:
Did you mean `pipelineMeta.ObjectMeta` and `pipelineSpec` could be nil? Both of them are from https://github.com/tektoncd/pipeline/blob/54c9b51295b2023e23f376128f90618381d927ac/pkg/reconciler/pipelinerun/pipelinespec/pipelinespec.go#L35-L38 
and they are initialized so I think they cannot be nil?

Reply:
That's not obvious from this func though 🙃 

I think you should change this to:

```go
if pipelineMeta != nil && pipelineSpec != nil {
	pipelineToBeVerified = &v1beta1.Pipeline{
		ObjectMeta: pipelineMeta.ObjectMeta,
		Spec:       *pipelineSpec,
	}
}
```

Reply:
Oh ok, I will update